Add New Relic Insights-based commands; update to remove requests to Servers API

acrewdson opened the pull request:
- (*update*: leaving this command temporarily disabled for now, see the comment in the code) adds an `insights` command for submitting arbitrary NRQL queries to the Insights API 
- updates the `servers` command to rename it `infra` and have it use Infrastructure data (via Insights) now that Servers is going away
- adds a `cfgov-deployed` command to show details of the most recent deployments for all cfgov environments 

For the `cfgov-deployed` command to work, we'll need to first merge the associated change to the cfgov Jenkins job (see GHE PR).

acrewdson committed:
Add 'insights' command to query the NR Insights API

New Relic Insights can be queried by submitting NRQL queries
to an API endpoint. This adds a new hubot command for doing that.

This first attempt supports regular queries that return lists
of events, along with faceted queries that group the results
by an attribute.

Other types of queries, including multi-facet queries, are not
yet supported.

acrewdson committed:
Update 'servers' commands in prep for NR Servers decom

New Relic's Servers API is going away, so we need to replace
the 'servers' commands, which rely on that API, with new
commands that query Infrastructure data via the Insights API.
This is a first attempt at doing that.

yunghuffy commented on src/newrelic.coffee:
```diff
@@ -52,6 +64,18 @@ plugin = (robot) ->
+    SELECT
+      average(cpuPercent) AS cpuPercent,
+      average(memoryUsedBytes / memoryTotalBytes) * 100 AS memoryPercent,
+      average(diskUsedPercent) as diskUsedPercent
```
I have a humble request to include memory used as a counting integer of MBs as well as percent.

acrewdson replied:
@champain here I was just attempting to match (to the extent possible) the existing `servers` output, which only has percentages; might be a separate PR to expand this command's output?

yunghuffy reviewed (dismissed):
Really great work. I think the NRQL stuff is a bit vague, and I wonder if that could be more intentional.

```
SELECT * FROM InfraEvents WHERE HOSTNAME = 'foo'
```
Something like the above somewhere to jog my memory on what NRQL is, or maybe just an embedded link to NRQL might help, especially if I'm asking the bot what it's expecting as an input for the `insights` command.

yunghuffy commented:
I'll try to spend some time with this locally for further review.

acrewdson committed:
Better handle situations where there is no JSON

acrewdson committed:
Add new 'cfgov-deployed' command

This is a highly-CFPB-specific command, `bot newrelic cfgov-deployed`,
that lists details about the last deploy (as recorded in New Relic
Insights) for all cfgov environments. The goal is to make the discovery
of this information as easy as possible.

acrewdson committed:
Disable the 'insights' command for now

We don't currently have enough confidence in the safety of this
command, so we will leave it disabled for now.

acrewdson committed:
Update README to reflect new/updated commands
